integration-tests/src/integration_test_circuits.rs

```diff
@@ -84,46 +91,203 @@ lazy_static! {
-    };
-    /// Tx Circuit proving key
-    pub static ref TX_CIRCUIT_KEY: ProvingKey<G1Affine> = {
-        let block = new_empty_block();
-        let circuit = TxCircuit::<Fr>::new_from_block(&block);
-        let general_params = get_general_params(TX_CIRCUIT_DEGREE);
-
-        let verifying_key =
-            keygen_vk(&general_params, &circuit).expect("keygen_vk should not fail");
-        keygen_pk(&general_params, verifying_key, &circuit).expect("keygen_pk should not fail")
-    };
-    /// Bytecode Circuit proving key
-    pub static ref BYTECODE_CIRCUIT_KEY: ProvingKey<G1Affine> = {
-        let block = new_empty_block();
-        let circuit = BytecodeCircuit::<Fr>::new_from_block(&block);
-        let general_params = get_general_params(BYTECODE_CIRCUIT_DEGREE);
-
-        let verifying_key =
-            keygen_vk(&general_params, &circuit).expect("keygen_vk should not fail");
-        keygen_pk(&general_params, verifying_key, &circuit).expect("keygen_pk should not fail")
-    };
-    /// Copy Circuit proving key
-    pub static ref COPY_CIRCUIT_KEY: ProvingKey<G1Affine> = {
-        let block = new_empty_block();
-        let circuit = CopyCircuit::<Fr>::new_from_block(&block);
-        let general_params = get_general_params(COPY_CIRCUIT_DEGREE);
-
-        let verifying_key =
-            keygen_vk(&general_params, &circuit).expect("keygen_vk should not fail");
-        keygen_pk(&general_params, verifying_key, &circuit).expect("keygen_pk should not fail")
-    };
+    /// Integration test for EVM circuit
+    pub static ref EVM_CIRCUIT_TEST: TokioMutex<IntegrationTest<EvmCircuit<Fr>>> =
+    TokioMutex::new(IntegrationTest::new("EVM", EVM_CIRCUIT_DEGREE));
+
+    /// Integration test for State circuit
+    pub static ref STATE_CIRCUIT_TEST: TokioMutex<IntegrationTest<StateCircuit<Fr>>> =
+    TokioMutex::new(IntegrationTest::new("State", STATE_CIRCUIT_DEGREE));
+
+    /// Integration test for State circuit
+    pub static ref TX_CIRCUIT_TEST: TokioMutex<IntegrationTest<TxCircuit<Fr>>> =
+    TokioMutex::new(IntegrationTest::new("Tx", TX_CIRCUIT_DEGREE));
+
+    /// Integration test for Bytecode circuit
+    pub static ref BYTECODE_CIRCUIT_TEST: TokioMutex<IntegrationTest<BytecodeCircuit<Fr>>> =
+    TokioMutex::new(IntegrationTest::new("Bytecode", BYTECODE_CIRCUIT_DEGREE));
+
+    /// Integration test for Copy circuit
+    pub static ref COPY_CIRCUIT_TEST: TokioMutex<IntegrationTest<CopyCircuit<Fr>>> =
+    TokioMutex::new(IntegrationTest::new("Copy", COPY_CIRCUIT_DEGREE));
+
+    /// Integration test for Copy circuit
+    pub static ref SUPER_CIRCUIT_TEST: TokioMutex<IntegrationTest<SuperCircuit::<Fr, MAX_TXS, MAX_CALLDATA, TEST_MOCK_RANDOMNESS>>> =
+    TokioMutex::new(IntegrationTest::new("Super", SUPER_CIRCUIT_DEGREE));
+}
+
+/// Generic implementation for integration tests
+pub struct IntegrationTest<C: SubCircuit<Fr> + Circuit<Fr>> {
```

Comment:
This looks very nice! :zap:

Reply:
Thanks!! Once we had to store the fixed columns, I think it made sense to abstract it out